conda/core/path_actions.py

```diff
@@ -1042,3 +1042,34 @@ def target_full_path(self):
     def __str__(self):
         return ('ExtractPackageAction<source_full_path=%r, target_full_path=%r>'
                 % (self.source_full_path, self.target_full_path))
+
+
+class DumpRecordAction(PathAction):
```

Comment:
Let's call it `DumpRepodataRecordAction`.  I looked back over that data model code email I sent you, and I think what we always want here *is* the RepodataRecord.

```python
class PackageRef(Entity):
    # fields important for uniquely identifying a package

    channel = ComposableField(Channel)
    subdir = StringField()
    filename = StringField()  # previously fn

    name = StringField()
    version = StringField()
    build_string = build = StringField()
    build_number = IntegerField()
    track_features = StringField(required=False)

    md5 = StringField()

class RepodataRecord(PackageRef):
    # important for "choosing" a package (i.e. the solver), listing packages
    # (like search), and for verifying downloads

    package_type = EnumField(NoarchType, required=False)  # previously noarch
    depends = ListField(string_types, required=False)
    constrains = ListField(string_types, required=False)
    features = ListField(string_types, required=False)
    preferred_env = StringField(required=False)
    size = IntegerField(required=False)
    license = StringField(required=False)
    license_family = StringField(required=False)
```

Right now there are extra fields like `leased_paths` and `preferred_env`, and that's only because we currently have a single `IndexRecord` data type that conflates `RepodataRecord` and `PrefixRecord`.